### PR TITLE
DROOLS-7220 Dispose rule unit session in executable model DMN

### PR DESCRIPTION
**Pull request by baldimir**
- Fixes a memory leak, when the RuleUnitExecutorSession was not being disposed. 
- I am sorry for not providing a test for this, however as this is a deprecated functionality in a legacy stream and having a test for this may be more complicated, I am submitting only a fix. If it will be required to provide a test, I could do it in a separate PR.

**JIRA**: https://issues.redhat.com/browse/DROOLS-7220

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-lts</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>

 - for <b>mandrel lts checks</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins run mandrel-lts</b>

- for a <b>specific mandrel lts check</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel-lts</b>
</details>

### CODE REVIEW

**Comment by baldimir** [comment]
Jenkins run cdb

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/execmodelbased/AbstractModelEvaluator.java [hunk 1]
```diff
@@ -74,39 +74,43 @@ public EvaluatorResult evaluate( DMNRuntimeEventManager eventManager, DMNResult
...
-                                                   EvaluatorResult.ResultType.FAILURE :
-                                                   EvaluatorResult.ResultType.SUCCESS );
-        } catch (RuntimeException e) {
-            logger.error(e.toString(), e);
-            throw e;
         } finally {
-            evalCtx.exitFrame();
-            DMNRuntimeEventManagerUtils.fireAfterEvaluateDecisionTable( eventManager, node.getName(), dTableModel.getDtName(), dmnResult,
-                    (eventResults != null ? eventResults.matchedRules : null), (eventResults != null ? eventResults.fired : null));
+            executor.dispose();
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/execmodelbased/AbstractModelEvaluator.java:107]: this `executor.dispose();` checks with me, but I'm not sure why you wanted overall to have this double-nested try-catch-finally block, please?

**Comment by baldimir** [comment]
Looks like compile downstream problem is unrelated: 

[ERROR] Failed to execute goal org.wildfly.plugins:wildfly-jar-maven-plugin:6.1.2.Final:package (default) on project business-monitoring-webapp: Packaging wildfly failed: Could not determine if the logging subsystem was present: WFLYCTL0202: No known child type named subsystem -> [Help 1]
[2022-10-31T17:16:04.270Z] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.wildfly.plugins:wildfly-jar-maven-plugin:6.1.2.Final:package (default) on project business-monitoring-webapp: Packaging wildfly failed

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/execmodelbased/AbstractModelEvaluator.java (hunk 1 above)
Comment (baldimir) [reply kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/execmodelbased/AbstractModelEvaluator.java:107]: I know it doesn't look nice, but I think that is the only way how to 100% guarantee that the session gets disposed, as there are these calls before it gets into the original try block: 

`EvaluationContext evalCtx = createEvaluationContext( events, eventManager, dmnResult );
evalCtx.enterFrame();`

Comment (baldimir) [reply kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/execmodelbased/AbstractModelEvaluator.java:107]: Hmm I realized I can reorder, the executor is only used later in the try block. So I will push a change.

Comment (baldimir) [reply kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/execmodelbased/AbstractModelEvaluator.java:107]: Done.

Comment (tarilabs) [reply kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/compiler/execmodelbased/AbstractModelEvaluator.java:107]: yes, or either way pushing the other lines down inside the `try` block as mentioned separately

**Review by tarilabs — APPROVED** [review]
this is much more minimal change and preferred, thanks

**Review by hellowdan — APPROVED** [review]

**Pull request merged**